### DISCUSSION TITLE
HTTP Error 400 when attempting to install a package

### DISCUSSION THREAD

**Discussion by sbrl** [atom/atom#23018] [Q&A] [answered]
Hello,

Today when I attempted to install a package in Atom (specifically `language-caddyfile`), I got a HTTP error 400:

```
npm ERR! code E400
npm ERR! 400 Bad Request - GET https://www.atom.io/api/packages/language-caddyfile/versions/0.1.7/tarball

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/sbrl/.atom/.apm/_logs/2021-09-20T20_31_30_359Z-debug.log
```
![Selection_2021-09-20_21:37:36_001_5b66eee](https://user-images.githubusercontent.com/9929737/134074121-63783e6b-38ff-40b1-993f-35c0ec80056f.png)

Anyone know what's up with that? It persists after a reload.

**Comment by heyhewmike** [comment, marked as answer]
If this is the package by @aaronellington, [atom.io package](https://atom.io/packages/language-caddyfile), seen on this link it appears this developer is no longer sharing the code repository on GitHub. If you go to the Repo link on the Atom.io page package page for [language-caddyfile](https://github.com/aaronellington/language-caddyfile) you will get a 404 error.

Since the Code Repository is no longer Public it is no longer accessible and you are getting the 400 Bad Request.

Edit: Tagging the GitHub user who's package I am referring to.

**Comment by aaronellington** [comment]
My apologies. The Github repository has been restored.

**Reply by sbrl** [reply]
Thanks so much! I'm just starting out with Caddy, and having an Atom grammar for it is essential I feel.

**Reply by aaronellington** [reply]
No worries! Just a disclaimer, I haven't updated the package in about 4 years and the package does not support Caddy v2. So you may run into some issues with it.

**Reply by sbrl** [reply]
Oh, I see. Thanks for the heads up! I was wondering why some of the syntax highlighting was a bit funky haha

Sometimes I feel like I'm the only GitHub Atom user left haha
